setup.py

```diff
@@ -14,22 +14,22 @@
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
+import os, sys
+import subprocess
+
 from setuptools import setup
 from setuptools.command.install import install
 from distutils.command.build import build
+from distutils.command.build_ext import build_ext
 from distutils.core import setup, Extension
-import os.path, sys
-import subprocess
 
-BASEPATH = os.path.dirname(os.path.abspath(__file__))
-FROTZPATH = os.path.join(BASEPATH, 'frotz')
-subprocess.check_call(['make', 'clean'], cwd=FROTZPATH)
-subprocess.check_call(['make', 'library', '-j', '4'], cwd=FROTZPATH)
 
-frotz_c_lib = 'jericho/libfrotz.so'
-if not os.path.isfile(frotz_c_lib):
-    print('ERROR: Unable to find required library %s.'%(frotz_c_lib))
-    sys.exit(1)
+class CustomBuildExt(build_ext):
+    def build_extension(self, ext):
+        FROTZPATH = os.path.join(os.getcwd(), 'frotz')
```

Comment:
I think you should keep `BASEPATH` around, in case `setup.py` is invoked from some other directory.

Reply:
I've only tried with `pip` and from what I understand it handles changing directory appropriately. But you are right, I haven't thought of calling `python setup.py` from another directory. I'll make the change, thanks.